cmd/cmd.go

```diff
@@ -35,6 +40,27 @@ import (
 	"github.com/spf13/cobra"
 )
 
+const (
+	dotGauge        = ".gauge"
+	prevCmdFileName = "prevCmd.json"
+)
+
+type prevCommand struct {
+	Command []string
+}
+
+func newPrevCommand() *prevCommand {
+	return &prevCommand{Command: make([]string, 0)}
+}
+
+func getJSON(failedMeta *prevCommand) string {
+	j, err := json.MarshalIndent(failedMeta, "", "\t")
+	if err != nil {
+		logger.Warningf("Failed to save run info. Reason: %s", err.Error())
```

Comment:
Doesn't look like the method is saving the run info here. Instead we can return the error here. and use the appropriate log message where needed.